File: Swift-C++-Testsuite/test/concurrency/Swift/BasicActors.swift
@main
struct MyMain {

    static func main() async {
        let logger = TemperatureLogger(label: "Outdoors", measurement: 25)
        let maxTemp = await logger.max
    }

}

actor TemperatureLogger {
    let label: String
    var measurements: [Int]
    private(set) var max: Int

    init(label: String, measurement: Int) {
        self.label = label
        self.measurements = [measurement]
        self.max = measurement
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/BasicActors.cpp

#include <iostream>
#include <string>
#include <vector>

class TemperatureLogger {
    public:
        TemperatureLogger(const std::string &label, int measurement) 
        : label(label) {
            measurements.push_back(measurement);
            max = measurement;
        }

        std::string getLabel() {
            return label;
        }

        std::vector<int> getMeasurements(){
            return measurements;
        }

        int getMax() {
            return max;
        }

    private:
        std::string label;
        std::vector<int> measurements;
        int max;
};


int main() {
    TemperatureLogger logger("Outdoors", 25);
    int maxTemp = logger.getMax();
    return 0;
}